Re-org: Rename plugin/ files and classes

JohannesLorenz opened the pull request:
Re-org PR! (#5592)

**Description**

This fixes casing of everything in plugins/ that is currently lowercase but should be uppercase.
Additional fixups:

* style.css: Fix opl2 -> Opulenz
* bs/sr.ts: Fix outdated entries

**Review**

This PR changes only code, i.e. no library names or strings for savefiles/presets. There are changes in the locales and themes though, but I verified all of them using scripts.

* Code review: I checked all files, but it can't be harm to have someone cross-check it.
* Test review: Check that
  - most native plugins still work (maybe check songs with many different native plugins)
  - most native plugins still look correct (no style issues)
  - most tooltips still work (and are translated, if they were before)

**Open TODO** (but it can be done later)

Remove duplicate translations. The following warnings show up while compiling:

```
Warning: dropping duplicate messages in '/home/johannes/cprogs/lmms/master/data/locale/es.qm':

* Context: PatchesDialog
* Source: Patch

* Context: PatchesDialog
* Source: Program selector

* Context: PatchesDialog
* Source: Bank

* Context: PatchesDialog
* Source: Bank selector

* Context: PatchesDialog
* Source: Qsynth: Channel Preset

* Context: PatchesDialog
* Source: Cancel

* Context: PatchesDialog
* Source: OK

* Context: PatchesDialog
* Source: Name
```

JohannesLorenz committed:
Fix casing of filenames and code in plugins/

No functional changes! No changes to savefiles/presets!

Fixes casing of everything that is currently lowercase but should
be uppercase.

Fixes also some other plugin strings, especially:

* opl2 -> OpulenZ (see 289887f4fc9bc5bce3678bf7aaa842fd19eba08c)
* calf -> veal (see ae291e0709b544bc51a831ea606211801fac14b6)
* ladspa_effect -> LadspaEffect (see 9c9372f0c87f6d3e6d5b0de3e5509bafa1da9687)
* remove flp_import (see 2d1813fb64275317d0a8cc19dbdd2300a1ede0a3)

JohannesLorenz commented on plugins/Sf2Player/Sf2Player.cpp:
```diff
@@ -23,7 +23,7 @@
  */
 
-#include "sf2_player.h"
+#include "Sf2Player.h"
```
I do what @tresf said :laughing: And he is right IMO. If I look into cmake/modules/PluginList.cmake, we did use `Vst`, `Lv2`, `Midi` etc. The `SF2` in the class names is an error that I need to fix.

JohannesLorenz commented on plugins/Sf2Player/PatchesDialog.cpp:
```diff
@@ -1,5 +1,5 @@
 /*
- * patches_dialog.cpp - display sf2 patches
+ * PatchesDialog.cpp - display sf2 patches
```
Sounds good. Will do.

JohannesLorenz commented on plugins/Lb302/Lb302.cpp:
```diff
@@ -1,14 +1,14 @@
 /*
- * lb302.cpp - implementation of class lb302 which is a bass synth attempting
+ * Lb302.cpp - implementation of class LB302 which is a bass synth attempting
```
Lb302, after the mentioned discussion :grinning:

JohannesLorenz replied:
Fixed, but only in those 2 file comments. All other occurrences seemed to be not the class name, but the instrument name, so LB302 seems correct in the other cases.

Veratil reviewed (approved):
LGTM!

JohannesLorenz commented:
> LGTM!

Perfect. Thanks for the 2 reviews!

Now the PR is open for voluntary testing. I'll merge it on the weekend if no one volunteers.

DomClark commented on cmake/modules/PluginList.cmake:
```diff
@@ -100,7 +100,7 @@ ENDIF()
 	SET(MSVC_INCOMPATIBLE_PLUGINS
 		LadspaEffect
-		zynaddsubfx
+		Zynaddsubfx
```
```suggestion
		ZynAddSubFx
```

DomClark commented on .gitmodules:
```diff
@@ -5,7 +5,7 @@
 	path = src/3rdparty/rpmalloc/rpmalloc
 	url = https://github.com/mjansson/rpmalloc.git
 [submodule "plugins/zynaddsubfx/zynaddsubfx"]
-	path = plugins/zynaddsubfx/zynaddsubfx
+	path = plugins/ZynAddSubFx/zynaddsubfx
```
Should the case of these paths match?

Veratil replied:
They would matter on Linux/Mac, but I think on Windows it ignores case. Good catch!

You'll also wants to updated the `[submodule "plugins/ZynAddSubFx/zynaddsubfx"]` section header.

JohannesLorenz commented:
@DomClark Thanks for the review. Good found x2. I found occurrences in even more files:

```
.gitignore
.gitmodules
cmake/modules/CheckSubmodules.cmake
cmake/modules/PluginList.cmake
debian/copyright
debian/patches/build-amd64-20181013.patch
doc/Doxyfile.in
```

Before I commit, I'll now check these files for other plugins and check the whole file tree for more files.

One thing where I'm not sure:

https://github.com/LMMS/lmms/blob/9628a2b512dba7086532bad2adee618bffa9fb53/CMakeLists.txt#L677-L686

Should I rename `plugins/zynaddsubfx` to `plugins/ZynAddSubFx` here, too? I don't know where these `extra/` files for this "win 32 build" come from. The last commit really dealing with it is eb2f5e9cc2 , though @jasp00 has also committed in that area (5b2283feec).

JohannesLorenz commented:
I pushed the fixes, and also checked for all other files in the LMMS tree. For the strings that I changed in the patches, I verified all new locations with a script.

As discussed in Discord, I removed the CMakeLists.txt "package ZynAddSubFX into win32 build" section. @DomClark , please verify the Windows installer.

JohannesLorenz commented:
Thanks a lot you two. I will merge tomorrow.
